lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -271,10 +272,14 @@ public function getInsertDiff(): array
     /**
      * INTERNAL: Gets the association mapping of the collection.
      *
-     * @psalm-return array<string, mixed>|null
+     * @psalm-return array<string, mixed>
      */
-    public function getMapping(): ?array
+    public function getMapping(): array
     {
+        if ($this->association === null) {
+            throw new UnexpectedValueException('The underlying association mapping is null although it should not be');
+        }
+
```

Comment:
Nice! Now you should use `getMapping()` where you previously added the null checks.